File: homeworks/lesson3/Figures/Rectangle.java
package homeworks.lesson3.Figures;

import homeworks.lesson3.Exceptions.IncorrectTriangleException;
import homeworks.lesson3.Exceptions.NegativeValueException;
import homeworks.lesson3.Figures.Base.Polygonal;

public class Rectangle extends Polygonal {

    public Rectangle(String sideA, String sideB) throws NegativeValueException, IncorrectTriangleException{
        super(new String[] { sideA, sideB, sideA, sideB });
    }

    public Rectangle() throws NegativeValueException, IncorrectTriangleException{
        this("3", "4");
    }

    @Override
    public double area() {
        return Double.parseDouble(sides[0]) * Double.parseDouble(sides[1]);
    }
    

    @Override
    public String toString() {
        return String.format("Прямоугольник\nСторона a: %s\nСторона b: %s", sides[0], sides[1]);
    }
}

File: homeworks/lesson6/Figures/Rectangle.cs
public class Rectangle : Polygonal
{
    public Rectangle(int a, int b) : base(new int[] {a, b, a, b})
    {

    }

    public Rectangle() : base(new int[] {2, 3, 2, 3})
    {
        
    }

    public override double area()
    {
        return base.sides[0] * base.sides[1];
    }

    public override string ToString()
    {
        return "Прямоугольник\nСторона a: " + base.sides[0] + "\nСторона b: " + base.sides[1];
    }
}